src/lib/routers/history.ts

```diff
@@ -123,8 +133,12 @@ class BrowserHistory<TRouteState> implements Router<TRouteState> {
 
       this.writeTimer = setTimeout(() => {
         setWindowTitle(title);
-        if (this.shouldPushState) {
+        if (
+          this.shouldPushState &&
+          this.lastHistoryLength === window.history.length
```

Comment:
this likely makes sense, but I wonder if we should have this condition only if InstantSearch is already unmounted?

It makes sense also to keep the length in all cases, even when we later expose pushState to different routers (for eg. vue router), as they would behave the same.

Just leaving this as a possibility, I think it makes sense to check in all cases

Reply:
If people elect to use the history router for IS, I think it's ok if our behaviour takes precedence during the lifecycle of InstantSearch.

For that, I agree we could only check the history length if `dispose()` has been called, so we're not losing state to url sync.

Reply:
yes, agreed, we probably want to enforce this bail out only happens after dispose. Maybe the condition could be rolled into the the shouldPushState value?

```
dispose() 
  this.shouldPushState = this.shouldPushState && this.lastHistoryLength === window.history.length
```

something like that maybe?